### PR TITLE
TRUNK-5816: Migrate api package tests to Junit5 4

### PR DESCRIPTION
**Pull request by achilep**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5816
## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by teleivo** [comment]
@achilep I have a fix over here https://github.com/teleivo/openmrs-core/commit/bdb77367798e3b97eb17fd9c0a4fef6b5b826576

**Comment by teleivo** [comment]
@achilep can you retrigger the build. am unsure if `getAllUsers_shouldNotContainsAnyDuplicateUsers` if flaky

File: api/src/test/java/org/openmrs/api/UserServiceTest.java [hunk 1]
```diff
@@ -11,13 +11,12 @@
...
-import static org.junit.jupiter.api.Assertions.assertNotSame;
-import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertThrows;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/UserServiceTest.java:19]: You seem to be going back to JUnit 4. Was that intentional?

Comment (achilep) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java:19]: yes is to correct this Error `UserServiceTest.getAllUsers_shouldNotContainsAnyDuplicateUsers:680 expected: <11> but was: <12>`  
for now

Comment (teleivo) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java:19]: Do you mean that we get this error on JUnit 5 but not JUnit 4?

**Comment by achilep** [comment]
Yes

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 2]
```diff
@@ -2268,6 +2238,9 @@ public void getOrders_shouldreturnDiscontinuedOrders() {
 		assertEquals(2, orders.size());
 		for (Order order : orders) {
 			assertEquals(DISCONTINUE, order.getAction());
+
+			assertTrue(order.getAction() == Action.DISCONTINUE);
+
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:2243]:
```suggestion
```

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 3]
```diff
@@ -2215,7 +2181,10 @@ public void getOrders_shouldreturnOrdersWithFulfillerStatusCompleted() {
 		List<Order> orders = orderService.getOrders(orderSearchCriteria);
 		assertEquals(1, orders.size());
 		for (Order order : orders) {
+
 			assertEquals(COMPLETED, order.getFulfillerStatus());
+
+			assertTrue(order.getFulfillerStatus() == Order.FulfillerStatus.COMPLETED);
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:2187]:
```suggestion
```

File: api/src/test/java/org/openmrs/api/OrderServiceTest.java [hunk 4]
```diff
@@ -2215,7 +2181,10 @@ public void getOrders_shouldreturnOrdersWithFulfillerStatusCompleted() {
 		List<Order> orders = orderService.getOrders(orderSearchCriteria);
 		assertEquals(1, orders.size());
 		for (Order order : orders) {
+
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/api/OrderServiceTest.java:2184]:
```suggestion
```

File: api/src/test/java/org/openmrs/api/UserServiceTest.java (hunk 1 above)
Comment (teleivo) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java:19]: please remove this file from the PR. tests for it on JUnit 5 are green on Bamboo and the Travis CI build on master https://travis-ci.org/github/openmrs/openmrs-core/builds/708421471

Comment (teleivo) [reply api/src/test/java/org/openmrs/api/UserServiceTest.java:19]: also see my suggestions you added some lines that have to be removed.

**Pull request merged**